src/cereal/schema_enum.hpp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <sstream>
+#include <iterator>
+#include <cereal/cereal.hpp>
+
+//#ifndef SCHEMA_ENUM 
+#define SCHEMA_ENUM(Name, ...) \
+struct Name \
+{ \
+    static const char* name() { return #Name; } \
+    \
+    static const std::vector<std::string>& strings() \
+    { \
+        static std::vector<std::string> v;  \
+        if (v.empty()) \
+        { \
+            std::string text = #__VA_ARGS__; \
+            std::replace(text.begin(), text.end(), ',', ' '); \
+            std::istringstream iss{text}; \
+            v = std::vector<std::string>{ \
+                std::istream_iterator<std::string>{iss}, \
+                std::istream_iterator<std::string>()}; \
+        } \
+        return v; \
+    }; \
+    static std::string verify(const std::string& candidate) \
+    { \
+        return (std::find(strings().begin(), strings().end(), candidate) == strings().end()) ? \
+            std::string{} : candidate; \
+    }\
+};
+
+//#endif SCHEMA_ENUM 
+
+namespace cereal {
+
+// typically instantiated with an enum created by the SCHEMA_ENUM macro
+template<typename SchemaEnum>
+class schema_enum_value
+{
+public:
+    static schema_enum_value from_string(const std::string& value)
+    {
+        schema_enum_value  ssv;
+        ssv.v_ = SchemaEnum::verify(value);
+        return ssv;
+    }
+
+    schema_enum_value& operator=(const std::string& v)
+    {
+        *this = from_string(v);
+        return *this;
+    }
```

Comment:
That reminds me, there are a couple of convenience enhancements I want to copy out of the prototype (most notably, a simple "operator==" implementation for direct comparisons with const std::string references)

Reply:
Sounds good to me!